home_4/src/builder/factory/AbstractShipmentBuilderFactory.ts

```diff
@@ -0,0 +1,6 @@
+import { Shipment } from "../../Shipment";
+import { ShipmentBuilder } from "../ShipmentBuilder";
+
+export abstract class AbstractShipmentBuilderFactory {
```

Comment:
just a flavor: if there is no plan to create any shareable methods between derived classes - it's better use interface - but it's not really required